internal/service/batch/job_definition_data_source.go

```diff
@@ -192,6 +195,44 @@ func (d *jobDefinitionDataSource) Read(ctx context.Context, request datasource.R
 		}
 	}
 
+	if cProps := jd.ContainerProperties; cProps != nil {
+		// HACK: populate .ResourceRequirements using the .Memory and .Vcpus fields.
+		// Currently, findJobDefinition() returns the deprecated fields and doesn't
+		// populate the preferred .ResourceRequirements field.
```

Comment:
These values have been [deprecated by AWS](https://docs.aws.amazon.com/batch/latest/APIReference/API_ContainerProperties.html#Batch-Type-ContainerProperties-memory). That being said... 
1. they haven't been removed by AWS yet
2. the way the current schema lays them out is significantly nicer easier to use (`container_properties.memory` vs `container_properties.resource_requirements[*].memory`) 

So am OK with not changing this until we "have to"... Terraform could choose to leave the values in both locations /shrug